src/darwin/Framework/CHIP/MTROTAProviderDelegateBridge.mm

```diff
@@ -574,11 +597,18 @@ bool GetPeerNodeInfo(CommandHandler * commandHandler, const ConcreteCommandPath
                           CHIP_ERROR err = gOtaSender.PrepareForTransfer(fabricIndex, nodeId);
                           if (CHIP_NO_ERROR != err) {
                               LogErrorOnFailure(err);
-                              handler->AddStatus(cachedCommandPath, Protocols::InteractionModel::Status::Failure);
-                              handle.Release();
-                              return;
+                              if (err == CHIP_ERROR_BUSY) {
+                                  response.status = static_cast<OTAQueryStatus>(MTROTASoftwareUpdateProviderOTAQueryStatusBusy);
+                                  handler->AddResponse(cachedCommandPath, response);
+                                  handle.Release();
+                                  return;
+                              } else {
+                                  handler->AddStatus(cachedCommandPath, Protocols::InteractionModel::Status::Failure);
```

Comment:
Is there a reason here to not convert the CHIP_ERROR into a status instead of using a generic status?

Reply:
we already set the correct value in the response.status from the delegate for all query response status - avilable, protocol not supported and are handling the busy case above. everything else we just map to generic failure for now.

Reply:
My point is: Why not map the "the failure code that corresponds to the CHIP_ERROR value" instead of "generic failure"?  Then the callee can communicate more nuanced status if needed.

Reply:
let me check.

Reply:
didn't get around to this. can we skip this for now? will file an issue.

Reply:
Followup is fine, yes.  Please file.

Reply:
done

Reply:
This is not done, but please just make sure a follow-up is filed.

Reply:
Filed - https://github.com/project-chip/connectedhomeip/issues/24838